### PR TITLE
More restoration documentation

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @goderbauer

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/restoration_properties.dart [hunk 1]
```diff
@@ -17,6 +17,78 @@ import 'restoration.dart';
...
+///   }
+///
+///   void _incrementAnswer() {
+///     setState(() {
+///       // The current value of the property can be accessed and modified via
+///       // the value getter and setter.
+///       _answer.value++;
+///     });
+///   }
+///
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/restoration_properties.dart:47]: For completeness, the RestorableInt needs to be disposed in `dispose`. See also https://github.com/flutter/flutter/pull/63497.

File: packages/flutter/lib/src/widgets/restoration_properties.dart [hunk 2]
```diff
@@ -17,6 +17,78 @@ import 'restoration.dart';
...
+///   @override
+///   Widget build(BuildContext context) {
+///     return OutlineButton(
```
[the remark this answers is not in the document]
Comment (HansMuller) [reply packages/flutter/lib/src/widgets/restoration_properties.dart:50]: Yes, please use OutlinedButton

[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/restoration_properties.dart:50]: ...

[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/restoration_properties.dart:50]: i've filed https://github.com/flutter/flutter/issues/63510

File: packages/flutter/lib/src/widgets/restoration_properties.dart [hunk 3]
```diff
@@ -17,6 +17,78 @@ import 'restoration.dart';
...
+///     return Duration(microseconds: data as int);
+///   }
+///
+///   @override
+///   Object toPrimitives() {
+///     return value.inMicroseconds;
+///   }
+///
+///   @override
+///   bool get enabled => true;
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/widgets/restoration_properties.dart:87]: removed

**Review by goderbauer — APPROVED** [review]
LGTM, but cirrus is unhappy.

**Comment by goderbauer** [comment]
(PR triage fly-by comment): The analyzer's still unhappy.

**Pull request merged**